rules/template-indent.js

```diff
@@ -0,0 +1,167 @@
+'use strict';
+const stripIndent = require('strip-indent');
+const indentString = require('indent-string');
+const esquery = require('esquery');
+const {replaceTemplateElement} = require('./fix/index.js');
+
+const MESSAGE_ID_IMPROPERLY_INDENTED_TEMPLATE = 'template-indent';
+const messages = {
+	[MESSAGE_ID_IMPROPERLY_INDENTED_TEMPLATE]: 'Templates should be properly indented.',
+};
+
+/** @param {import('eslint').Rule.RuleContext} context */
+const create = context => {
+	const sourceCode = context.getSourceCode();
+	const options = {
+		tags: ['outdent', 'dedent', 'gql', 'sql', 'html', 'styled'],
+		functions: ['dedent', 'stripIndent'],
+		selectors: [],
+		comments: ['HTML', 'indent'],
+		...context.options[0],
+	};
+
+	options.comments = options.comments.map(comment => comment.toLowerCase());
+
+	const selectors = [
+		...options.tags.map(tag => `TaggedTemplateExpression[tag.name="${tag}"] > .quasi`),
+		...options.functions.map(fn => `CallExpression[callee.name="${fn}"] > .arguments:first-child`),
+		...options.selectors,
+	];
+
+	/** @param {import('@babel/core').types.TemplateLiteral} node */
+	const indentTemplateLiteralNode = node => {
+		const delimiter = '__PLACEHOLDER__' + Math.random();
+		const joined = node.quasis
+			.map((quasi, index) => {
+				const untrimmedText = sourceCode.getText(quasi);
+				const last = node.quasis.length - 1;
+				return untrimmedText.slice(1, index === last ? -1 : -2);
+			})
+			.join(delimiter);
+
+		const eolMatch = joined.match(/\r?\n/);
+		if (!eolMatch) {
+			return;
+		}
+
+		const eol = eolMatch[0];
```

Comment:
We already have configurable `indent`, do you think it's a good idea to allow config `eol` too?

Reply:
Bad idea, please ignore.